### ISSUE TITLE
If a client creates a lot of invoices it will get MaxSlotsExceeded

### ISSUE THREAD

**Issue by douglaz** [fedimint/fedimint#3406]
This client created some invoices but couldn't get them paid due to other issues, now it's getting this while creating new invoices:
```
$ sudo -u monitoring /run/current-system/sw/bin/fedimint-cli-rel-a06 --data-dir /home/monitoring/mutinynet-02 ln-invoice --amount 1024
2023-10-16T19:12:35.178598Z  INFO fedimint_client::sm::executor: Starting state machine executor task
2023-10-16T19:12:35.659869Z  WARN transaction_submission: Federation returned error error=FederationError { general: Some(Received errors from 2 peers: peer-0: Rpc error: Configured max number of request slots exceeded, peer-2: Rpc error: Configured max number of request slots exceeded), peers: {PeerId(0): Rpc(MaxSlotsExceeded), PeerId(2): Rpc(MaxSlotsExceeded)} }
2023-10-16T19:12:35.660619Z  WARN transaction_submission: Federation returned error error=FederationError { general: Some(Received errors from 2 peers: peer-0: Rpc error: Configured max number of request slots exceeded, peer-1: Rpc error: Configured max number of request slots exceeded), peers: {PeerId(0): Rpc(MaxSlotsExceeded), PeerId(1): Rpc(MaxSlotsExceeded)} }
(...)
```

**Comment by elsirion** [comment]
Can we just up that limit? I think you ran into that before when doing stress-testing.

**Comment by douglaz** [comment]
> Can we just up that limit? I think you ran into that before when doing stress-testing.

Yeah, perhaps it's the same problem. We could do that, but it still seems as a workaround, not as a proper solution.

**Comment by elsirion** [comment]
Imo having a limit for concurrent requests is silly. There is only one WS connection for all of them anyway, the overhead per request should be negligible.

With the Tor integration we need to separate out requests by operation id anyway, so I doubt a single JSON RPC client will still hit these limits.

**Comment by douglaz** [comment]
Oh, I thought it was a limit of connections, not of requests. So yes, there is no reason for it to be so low.

**Comment by justinmoon** [comment]
Call: This is a client-side limit. Increasing doesn't have a lot of overhead. @douglaz notes that this happened a few days after generating the invoice.

**Comment by justinmoon** [comment]
Should this error be retryable? [It currently is](https://github.com/fedimint/fedimint/blob/0ef5039086a4cd863484cbfacd638d8a0d42c233/fedimint-core/src/api.rs#L81-L98), but I think this just led to a silent failure that really confused us for a long time. We were debugging for like 3 hours before we noticed this was what was happening. Perhaps logging is the solution to this particular problem ...

**Comment by elsirion** [comment]
We set the maximum for parallel requests to `u16::MAX`. The action items I see left:

* Better logging when we run into this
* Being less wasteful with requests (tracked in #3741)

**Comment by elsirion** [comment]
> We set the maximum for parallel requests to `u16::MAX`. The action items I see left:
> 
>     * Better logging when we run into this
> 
>     * Being less wasteful with requests (tracked in [Combine e-cash state machines #3741](https://github.com/fedimint/fedimint/issues/3741))

Both proposed changes happened and we are switching to Iroh anyway
